### PR TITLE
Added `nakayoshi_fork` option.

### PR DESCRIPTION
**Pull request by nateberkopec**
Reduce memory usage in preloaded cluster-mode apps by GCing before
fork and compacting, where available.

### Description
Please describe your pull request. Thank you for contributing! You're the best.

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` the pull request title.
- [x] I have added appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

**Comment by jjb** [comment]
the nakayoshi gem does a max of 4 times, not sure if that's a theoretical ideal to also use here

**Comment by nateberkopec** [comment]
True, changed. I'm going to leave out the full algorithm for now because a) license issues w/incorporating someone else's MIT-licensed algo that I don't feel like dealing with and b) compact implies GC, so we could probably GC fewer times in that case.

I want to punt on all the perf issues and possible optimizations until we figure out if this even decreases memory usage at all, if it does then we can deal with these issues.

File: lib/puma/dsl.rb [hunk 1]
```diff
@@ -759,5 +759,18 @@ def set_remote_address(val=:socket)
...
+    # When enabled, Puma will GC 4 times before forking workers.
+    # If available (Ruby 2.7+), we will also call GC.compact.
+    # Not recommended for non-MRI Rubies.
+    #
+    # Based on the work of Koichi Sasada and Aaron Patterson, this option may
+    # decrease memory utilization of preload-enabled cluster-mode Pumas. It will
+    # also increase time to boot and fork. See your logs for details on how much
+    # time this adds to your boot process. For most apps, it will be less than one
+    # second.
+    def nakayoshi_fork(enabled=false)
```
Comment (schneems) [inline lib/puma/dsl.rb:772]: Why are we defaulting this to false? It seems weird that adding this in my config does nothing:

```
nakayoshi_fork
```

I have to 

```
nakayoshi_fork(true)
```

Was this intentional or can we change the default?

Comment (nateberkopec) [reply lib/puma/dsl.rb:772]: Thaaaaat would be a bug

Comment (schneems) [reply lib/puma/dsl.rb:772]: Maybe that’s why there was so little change in CodeTriage

**Pull request merged**